accepted/000070-cluster_awareness.md

```diff
@@ -0,0 +1,272 @@
+- Feature Name: MVP Cluster awareness for Uyuni/SUSE Manager
+- Start Date: 2020-02-20
+- RFC PR: https://github.com/uyuni-project/uyuni-rfc/pull/36
+
+# Summary
+[summary]: #summary
+
+Make Uyuni/SUSE Manager aware and capable of deploying and managing clusters.
+
+# Motivation
+[motivation]: #motivation
+
+<!-- - Why are we doing this?
+- What use cases does it support?
+- What is the expected outcome?
+
+Describe the problem you are trying to solve, and its constraints, without coupling them too closely to the solution you have in mind. If this RFC is not accepted, the motivation can be used to develop alternative solutions. -->
+
+Uyuni/SUSE Manager should be able to interact with other SUSE products:
+- SUSE Container as a Service Platform (CaaSP) and Cloud Application Platform (CAP). From now on, we consider CAP as an equivalent of CaaSP for our purposes
+- SUSE Enterprise Storage (SES)
+- SUSE Linux Enterprise High Availability (SLE-HA) and SUSE Linux Enterprise HA for SAP Products (SHAP)
+- Other types of clusters that might be released in the future
+
+The three SUSE products above constituted our initial study from which we tried to summarize the common actions that a user is interested to issue on a cluster, as much as generically speaking as possible.
+This RFC aims to explain how Uyuni/SUSE Manager can __initially__ support the idea of being aware of a cluster and issue a specific set of actions on it. This is the MVP for a cluster: future RFCs will be extending this RFC horizontally (more cluster actions) and vertically (extend an implemented action by adding optional specific actions).
+
+# Detailed design
+[design]: #detailed-design
+
+<!-- This is the bulk of the RFC. Explain the design in enough detail for somebody familiar with the product to understand, and for somebody familiar with the internals to implement.
+
+This section should cover architecture aspects and the rationale behind disruptive technical decisions (when applicable), as well as corner-cases and warnings. Whenever the new feature creates new user interactions, this section should include examples of how the feature will be used. -->
+
+## Assumptions
+
+- During the writing of this RFC, we considered the documentation available of the latest version of the following products: CaaSP 4.1, SES 6, SLE-HA 15 SP1
+- All products that want to integrate with Uyuni/SUSE Manager must provide a cluster provider manager in the form of a Salt module that implements the actions described below: add a node, remove a node, upgrade the cluster.
+- Managing an existing cluster has priority over deploying a new cluster from the Uyuni/SUSE Manager perspective
```

Comment:
I see your point.

All in all, I think the root issue is that we did not so far agree (in detail) on what should or should not be in an RFC - we do not have a "best practices guide" of some sort. FMPOV, and so far, SUSE internal prioritization considerations were not considered part of what is essentially a technical feedback soliciting document, moreover open to the community - but that's not set in stone.

I guess this could be discussed in future, in a general context.